src/checkqueue.h

```diff
@@ -145,12 +154,14 @@ class CCheckQueue
     //! Add a batch of checks to the queue
     void Add(std::vector<T>& vChecks)
     {
-        boost::unique_lock<boost::mutex> lock(mutex);
-        BOOST_FOREACH (T& check, vChecks) {
-            queue.push_back(T());
-            check.swap(queue.back());
+        {
```

Comment:
Just noting that there is a slight functional change here, the lock is released earlier but this seems a-OK.